src/backend/core/user/models.py

```diff
@@ -0,0 +1,131 @@
+from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+from django.conf import settings
+
+from cart.models import Cart
+
+from datetime import datetime
+
+import jwt
+
+
+# Manager for User model
+class UserManager(BaseUserManager):
+    def create_user(self, email, password=None):
+        if not email:
+            raise ValueError("Users must have an email address")
+        if not password:
+            raise ValueError("Users must have a password")
+
+        # check whether user already exists
+        if User.objects.filter(email=email).exists():
+            raise ValueError("User already exists")
+
+        user = self.model(
+            email=self.normalize_email(email),
+        )
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password):
+        if not email:
+            raise ValueError("SuperUsers must have an email address")
+        if not password:
+            raise ValueError("SuperUsers must have a password")
+
+        user = self.create_user(
+            email=self.normalize_email(email),
+            password=password,
+        )
+
+        user.is_admin = True
+        user.is_staff = True
+        user.save(using=self._db)
+        return user
+
+
+# User model
+class User(AbstractBaseUser):
+    email = models.EmailField(max_length=40, unique=True)
```

Comment:
Maybe let's make this Primary Key since it has unique constraint 🤔 ?

Reply:
Agreed.